src/raw_iter.rs

```diff
@@ -69,17 +69,372 @@ enum DynamicRawColors<'a> {
+                                        return None;
+                                    }
+                                }
+                                1 => {
+                                    // End of bitmap
+                                    self.rle_state = RleState::EndOfBitmap;
+                                }
+                                2 => {
+                                    // Delta
+                                    panic!("Delta is unsupported");
+                                }
+                                _ => {
+                                    // Absolute mode
+                                    self.rle_state = RleState::Absolute {
+                                        remaining: param.saturating_sub(1),
+                                        is_odd: (param % 2) != 0,
+                                        // Odd lengths in RLE8 require 1 byte padding
+                                        has_padding: (param % 2) != 0,
+                                    };
+                                }
+                            }
+                        }
+                        _ => {
+                            // An encoded run
+                            self.rle_state = RleState::Running {
+                                remaining: length.saturating_sub(1),
+                                value: param,
+                                is_odd: (length % 2) != 0,
+                            };
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Iterator over individual BMP RLE4 encoded pixels.
+///
+/// Each pixel is returned as a `u32` regardless of the bit depth of the source image.
+#[derive(Debug)]
+pub struct Rle4Pixels<'a> {
+    /// Our source data
+    data: &'a [u8],
+    /// Our state
+    rle_state: RleState,
+    /// The width of a line in pixels
+    width: u32,
+    /// The location of the next pixel
+    next_pixel: Point,
+}
+
+impl<'a> Rle4Pixels<'a> {
+    /// Create a new RLE pixel iterator.
+    pub(crate) fn new(raw_bmp: &RawBmp<'a>) -> Rle4Pixels<'a> {
+        let header = raw_bmp.header();
+        Rle4Pixels {
+            data: raw_bmp.image_data(),
+            rle_state: RleState::Starting,
+            width: header.image_size.width,
+            // RLE encoded bitmaps are upside down
+            next_pixel: Point::new(0, (header.image_size.height - 1) as i32),
+        }
+    }
+
+    /// Bump the cursor to the next position in the bitmap.
+    ///
+    /// Note that RLE bitmaps are upside down.
+    fn move_position(&mut self) -> Point {
+        let old_position = self.next_pixel;
+        self.next_pixel.x += 1;
+        if self.next_pixel.x == self.width as i32 {
+            self.next_pixel.x = 0;
+            self.next_pixel.y = self.next_pixel.y.saturating_sub(1);
+        }
+        old_position
+    }
+}
+
+impl<'a> Iterator for Rle4Pixels<'a> {
+    type Item = (Point, u32);
```

Comment:
The `RawPixel` type has the same fields as this tuple.
```suggestion
    type Item = RawPixel;
```